### PR TITLE
mds: set the proper extra bl for the create request

### PR DESCRIPTION
**Pull request by lxbsz**
Just in case a create request was totally completed but the client
didn't receive any unsafe and safe responses, and then when it's
retried in the MDS side it will be treated as a open request and
will miss setting the create ino for the reply message.

Finally for client it just sent a create request and then the MDS
just sent back a open reply, which will let the client miss setting
the CREATED flag and then the VFS will fail the create by returnning
a -EEXIST errno.

Fixes: https://tracker.ceph.com/issues/63265

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "quincy"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

- When filling out the below checklist, you may click boxes directly in the GitHub web UI.  When entering or editing the entire PR message in the GitHub web UI editor, you may also select a checklist item by adding an `x` between the brackets: `[x]`.  Spaces and capitalization matter when checking off items this way.

## Checklist
- Tracker (select at least one)
  - [x] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
- `jenkins test rook e2e`
</details>

### CODE REVIEW

**Review by gregsfortytwo — APPROVED** [review]
LGTM

File: src/mds/Server.cc [hunk 1]
```diff
@@ -2584,6 +2609,12 @@ void Server::handle_client_request(const cref_t<MClientRequest> &req)
   if (session) {
     mdr->session = session;
     session->requests.push_back(&mdr->item_session_request);
+
+    if (has_completed && req->get_op() == CEPH_MDS_OP_CREATE) {
+      ceph_assert(created != inodeno_t());
+
+      set_reply_extra_bl(req, created, mdr->reply_extra_bl);
```
Comment (batrick) [inline src/mds/Server.cc:2616]: I don't think this is the right place for this. It either needs to go in `handle_client_open` or in the switch statement in `Server::dispatch_client_request`.

**Review by batrick — CHANGES_REQUESTED** [review]
otherwise LGTM!

File: src/mds/Server.cc (hunk 1 above)
Comment (lxbsz) [reply src/mds/Server.cc:2616]: Okay, I will move it to the switch statement in `Server::dispatch_client_request()`. Thanks!

**Review by batrick — APPROVED** [review]

**Comment by batrick** [comment]
jenkins test make check arm64

**Comment by batrick** [comment]
This PR is under test in [https://tracker.ceph.com/issues/66433](https://tracker.ceph.com/issues/66433).

**Comment by batrick** [comment]
This PR is under test in [https://tracker.ceph.com/issues/66462](https://tracker.ceph.com/issues/66462).

**Comment by batrick** [comment]
This PR is under test in [https://tracker.ceph.com/issues/66609](https://tracker.ceph.com/issues/66609).

**Comment by batrick** [comment]
https://tracker.ceph.com/issues/66609#note-1

**Pull request merged**